pkg/apis/core/v1/validation/validation.go

```diff
@@ -42,18 +43,18 @@ func ValidateResourceRequirements(requirements *v1.ResourceRequirements, fldPath
 	for resourceName, quantity := range requirements.Limits {
 		fldPath := limPath.Key(string(resourceName))
 		// Validate resource name.
-		allErrs = append(allErrs, ValidateContainerResourceName(string(resourceName), fldPath)...)
+		allErrs = append(allErrs, ValidateContainerResourceName(core.ResourceName(resourceName), fldPath)...)
```

Comment:
You should not need the cast to `core.ResourceName` here, right?   Unless I am misreading, it is already `ResourceName` type

Reply:
They are from different packages😢
```
v1 "k8s.io/api/core/v1"
"k8s.io/kubernetes/pkg/apis/core"
```

```
cannot use resourceName (variable of type "k8s.io/api/core/v1".ResourceName) as core.ResourceName value in argument to ValidateContainerResourceName
```

Reply:
ugh, I hate stuff like this.  internal version is so clunky.